python/pyspark/testing/utils.py

```diff
@@ -758,11 +758,14 @@ def assertDataFrameEqual(
     has_pandas = False
     try:
         # If pandas dependencies are available, allow pandas or pandas-on-Spark DataFrame
-        import pyspark.pandas as ps
         import pandas as pd
         from pyspark.testing.pandasutils import PandasOnSparkTestUtils
 
         has_pandas = True
+        if has_pandas:
+            # Only import pyspark.pandas when Pandas is available to prevent test stopping
+            # when Pandas is not installed.
+            import pyspark.pandas as ps
```

Comment:
ps reference below will fail. Can we work around

Reply:
relocated the import